### ISSUE TITLE
Ajax validation example broken in Docs

### ISSUE THREAD

**Issue by xsanch** [django-crispy-forms/django-crispy-forms#280]
Hello,

I am trying the AJAX example from the documentaion anyway when I render the form with the render_crispy_form I am missing the CSRF token in the result HTML . Is there any way how to resolve this ? 

<code>
Here is the view:
@login_required
@json_view
def save_profileform(request):
        form = userprofileform(request.POST or None,request=request)
        if form.is_valid():
                form.save()
                return {'success': True}
        form_html = render_crispy_form(form)
        return {'success': False, 'form_html': form_html}
</code>

**Comment by fpghost** [comment]
The approach of @pydolan did not work for me with Django 1.8. Doing `ctx=RequestContext(request)`, did not produce a context with the csrf token key in the dictionary (even with  `django.middleware.csrf.CsrfViewMiddleware` in `MIDDLEWARE_CLASSES`, and with `django.template.context_processors.csrf` added to the `context_processors`option of `TEMPLATES`).

 I'm not sure if things changed in Django 1.8, but you can see in the `__init__` method of the RequestContext object that by default, `_processors` is set to None, and it's only upon `bind_to_template` that you would get csrf key added to the dictionary. 

Instead I found that
            ctx = {}
            ctx.update(csrf(request))
            form_html = render_crispy_form(form, context=ctx)

worked in my view.

**Comment by vehsakul** [comment]
@fpghost I have the same issue. Your solution works.

**Comment by carltongibson** [comment]
[From the docs](http://django-crispy-forms.readthedocs.org/en/latest/crispy_tag_forms.html?highlight=render_crispy_form#render-a-form-within-python-code): 

> Remember to pass your CSRF token to the helper method using the context dictionary...

This is taken care of with the template tag but, yeah, you need to do it manually if your going to render the form yourself. 

Under the hood crispy forms just uses the `csrf_token` template tag — which just pulls it from the context — IIRC you get a warning in DEBUG if this isn't set...

**Comment by fpghost** [comment]
@carltongibson Yes, but those same docs suggest using `request_context = RequestContext(request)` and then `render_crispy_form(form, context=request_context)` as the means to passing the CSRF token, which does not seem to work. I'm not sure why it doesn't work, but it is plain to see from django code that RequestContext when initialized (__init__) doesn't get the CSRF added, it's only on the `bind_to_template` method of RequestContext when called that it actually gets processors and CSRF added.. (you can just debug print the `request_context` before passing to `render_crispy_form` to verify this).Not sure if this is new behaviour in Django 1.8 or not.  Either way the manual method of adding context works, but it would nice to get to the bottom of this.

**Comment by carltongibson** [comment]
Ah, I see that — so is the Ajax validation example in the docs broken now?

**Comment by vehsakul** [comment]
@carltongibson Yes

**Comment by carltongibson** [comment]
OK — PRs fixing that example __very welcome__ :)

**Comment by fpghost** [comment]
@carltongibson  OK PR sent.

**Comment by carltongibson** [comment]
Closed by #523

**Comment by pbt1v** [comment]
Sorry for the necro -- I was trying to follow docs and the AJAX form validation kept tripping the clickjacking middleware.

I found the code [here](https://github.com/django-crispy-forms/django-crispy-forms/issues/689#issue-205586565) to solve my problem:

   ```
 form = UserForm(request.POST or None)
    if form.is_valid():
        form.instance.user = request.user
        form.save()
         # do something
    ctx = {}
    ctx.update(csrf(request))

    # needed to include the form=form.helper kwarg
    form_html = render_crispy_form(form, helper=form.helper,context=ctx)
    #return new form

```
